engine/access/rpc/backend/backend.go

```diff
@@ -355,6 +355,40 @@ func (b *Backend) GetLatestProtocolStateSnapshot(_ context.Context) ([]byte, err
 	return convert.SnapshotToBytes(validSnapshot)
 }
 
+// GetProtocolStateSnapshotByBlockID returns finalized snapshot by block id
+func (b *Backend) GetProtocolStateSnapshotByBlockID(_ context.Context, blockID flow.Identifier) ([]byte, error) {
```

Comment:
we should only need to implement these once. I think you can remove them from this file, and just use the version in `backend_network.go`. You can also delete the `GetLatestProtocolStateSnapshot` from this file as well